agent/src/main.rs

```diff
@@ -845,23 +921,36 @@ fn worker(
                     info!(log, "SMF ok!");
                 }
 
-                // After SMF successfully shuts off downstairs, remove zfs
-                // dataset.
-                let region_dataset =
-                    regions_dataset.from_child_dataset(&r.id.0).unwrap();
+                // `apply_smf` returned Ok, so the desired state transition
+                // succeeded: update the datafile.
+                let res = match &rs.state {
+                    State::Requested => {
+                        df.created_rs(&region_id, &snapshot_name)
+                    }
+
+                    State::Tombstoned => {
+                        df.destroyed_rs(&region_id, &snapshot_name)
+                    }
 
-                let res = worker_region_destroy(&log, &r, region_dataset)
-                    .and_then(|_| df.destroyed(&r.id));
+                    _ => {
+                        eprintln!(
```

Comment:
should we be using `error!` macro here instead of `eprintln!`?

There is another `eprintln!` down below as well that we should probably change.